rs/execution_environment/src/execution_environment/tests/canister_task.rs

```diff
@@ -819,3 +820,253 @@ fn global_timer_produces_transient_error_on_out_of_cycles() {
 
     assert_eq!(RejectCode::SysTransient, err.code().into());
 }
+
+#[test]
+fn on_low_wasm_memory_is_executed() {
+    let mut test = ExecutionTestBuilder::new().build();
+
+    let wat = r#"(module
+            (import "ic0" "msg_reply" (func $msg_reply))
+            (func $grow_mem
+                (drop (memory.grow (i32.const 7)))
+                (call $msg_reply)
+            )
+            (export "canister_update grow_mem" (func $grow_mem))
+            (func (export "canister_on_low_wasm_memory")
+                (drop (memory.grow (i32.const 5)))
+            )
+            (memory 1 20)
+        )"#;
```

Comment:
This wat file is basically the same across multiple tests, I think pulling it out in a const (or even some formatted string where you can vary how many pages are grown in `grow_mem`) will help keep the tests smaller. Did you look into the [WatCanisterBuilder](https://sourcegraph.com/github.com/dfinity/ic/-/blob/rs/test_utilities/execution_environment/src/wat_canister.rs?L292) in case it helps with better readability of these tests instead of dropping wat files?

Reply:
I refactor the creation of the wat file. I took a look at WatCanisterBuilder but I think it will take more time and that the benefit will not pay off.